cabot/plugin_test_utils.py

```diff
@@ -0,0 +1,230 @@
+            max_build_failures=5
+        )
+
+        self.http_check = HttpStatusCheck.objects.create(
+            id=10102,
+            name='Http Check',
+            created_by=self.user,
+            importance=Service.CRITICAL_STATUS,
+            endpoint='https://google.com',
+            timeout=10,
+            status_code='200',
+            text_match=None,
+        )
+        self.tcp_check = TCPStatusCheck.objects.create(
+            id=10103,
+            name='TCP Check',
+            created_by=self.user,
+            importance=Service.ERROR_STATUS,
+            address='github.com',
+            port=80,
+            timeout=6,
+        )
+
+        self.service = Service.objects.create(id=2194, name='Service')
+
+        self.service.status_checks.add(
+            self.jenkins_check,
+            self.http_check,
+            self.tcp_check
+        )
+
+        self.service.users_to_notify.add(self.user)
+
+        self.duty_officer = User.objects.create(
+            username='dolores@affirm.com',
+            password='fakepassword',
+            email='dolores@affirm.com',
+            is_active=True
+        )
+        self.fallback_officer = User.objects.create(
+            username='teddy@affirm.com',
+            password='fakepassword',
+            email='teddy@affirm.com',
+            is_active=True
+        )
+
+        self.schedule = Schedule.objects.create(name='Test Schedule', fallback_officer=self.fallback_officer)
+        Shift.objects.create(user=self.duty_officer,
+                             start=timezone.now() - timezone.timedelta(minutes=1),
+                             end=timezone.now() + timezone.timedelta(hours=1),
+                             schedule=self.schedule)
+        self.service.schedules.add(self.schedule)
+
+        self.assertEquals(get_duty_officers(self.schedule), [self.duty_officer])
+        self.assertEquals(get_fallback_officers(self.schedule), [self.fallback_officer])
+
+        self.service.update_status()  # set initial status
+
+        # make sure all users have profiles
+        for user in self.user, self.duty_officer, self.fallback_officer:
+            UserProfile.objects.create(user=user)
+
+        super(PluginTestCase, self).setUp()
+
+    def run_checks(self, checks, from_service_status=None):
+        # type: (List[Tuple[StatusCheck, bool, bool]], Union[None, str]) -> None
+        """
+        Simulates running the given checks with the given results, then updates the service (triggering alerts).
+        :param checks: list of (check, succeeded, acked) tuples
+        :param from_service_status: specify the service status to transition from (service.old_overall_status), optional
+        """
+        # clear any previous results
+        StatusCheckResult.objects.all().delete()
+
+        time_per_run = timezone.timedelta(seconds=1)
+        start = timezone.now() - time_per_run * (len(checks) + 1)
+        finish = start + time_per_run
```

Comment:
What's the point of all this time logic? (or what breaks if we just set all these values to some random time?)

Reply:
There's some logic in acks that checks `result.time_completed` again `ack.expire_at` and `ack.closed_at` - but those should only matter if the results are in the future. I'll just set both to `timezone.now()`, which should work just as well.